src/cfnlint/rules/resources/ectwo/RouteTableAssociation.py

```diff
@@ -0,0 +1,86 @@
+"""
+  Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+  Permission is hereby granted, free of charge, to any person obtaining a copy of this
+  software and associated documentation files (the "Software"), to deal in the Software
+  without restriction, including without limitation the rights to use, copy, modify,
+  merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
+  permit persons to whom the Software is furnished to do so.
+
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+  INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+  PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+  HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+  OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+  SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+"""
+from cfnlint import CloudFormationLintRule
+from cfnlint import RuleMatch
+import cfnlint.helpers
+
+
+class RouteTableAssociation(CloudFormationLintRule):
+    """Check only one route table association defined per subnet"""
+    id = 'E3022'
+    shortdesc = 'Resource SubnetRouteTableAssociation Properties'
+    description = 'Validate there is only one SubnetRouteTableAssociation per subnet'
+    source_url = 'https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-subnet-route-table-assoc.html'
+    tags = ['resources', 'subnet', 'route table']
+
+    # Namespace for unique associated subnets in the form condition::value
+    associated = set()
+
+    def get_values(self, subnetid, condition):
+        """Get string literal(s) from value of SubnetId"""
+        values = []
+        if isinstance(subnetid, dict):
+            if len(subnetid) == 1:
+                for key, value in subnetid.items():
+                    if key in cfnlint.helpers.CONDITION_FUNCTIONS:
+                        if isinstance(value, list):
+                            if len(value) == 3:
+                                condition = value[0]
+                                values.extend(self.get_values(value[1], condition))
+                                values.extend(self.get_values(value[2], condition))
+                    if key in ('Ref', 'GetAtt'):
+                        values.extend(self.get_values(value, condition))
+        else:
```

Comment:
is this else is important.  I'm trying to run scenarios in my head related to Conditions (and lets be clear condition handling is hard especially across resources).  What may be more appropriate is a bale out when things get too crazy.    The scenario here I'm wondering about is what happens if the Resource has a condition and property SubnetId has a condition.  In this case the subnetid could be a dict and a condition parameter was passed in but would never get added to values because of the else.

Reply:
I actually talked to Chuck offline about this and recommended that Conditions just be taken out of the picture. Just look for duplicates on the inline string/ref target and leave it at that. It's a small step for man.